client/src/unifycr-sysio.c

```diff
@@ -789,6 +789,43 @@ int UNIFYCR_WRAP(open64)(const char *path, int flags, ...)
     return ret;
 }
 
+int UNIFYCR_WRAP(__open_2)(const char* path, int flags, ...)
+{
+    int ret;
+
+    DEBUG("__open_2 was called for path %s....\n", path);
+
+    /* if O_CREAT is set, we should also have some mode flags */
+    int mode = 0;
+    if (flags & O_CREAT) {
+        va_list arg;
+        va_start(arg, flags);
+        mode = va_arg(arg, int);
+        va_end(arg);
+    }
+
+    /* check whether we should intercept this path */
+    if (unifycr_intercept_path(path)) {
+        DEBUG("__open_2 was intercepted for path %s....\n", path);
+
+        /* Call open wrapper with LARGEFILE flag set*/
+        if (flags & O_CREAT) {
+            ret = UNIFYCR_WRAP(open)(path, flags | O_LARGEFILE, mode);
+        } else {
+            ret = UNIFYCR_WRAP(open)(path, flags | O_LARGEFILE);
```

Comment:
Checking this one more time, I believe we want to drop the O_LARGEFILE flag here and update the comment about LARGEFILE.  That flag is just for open64, and I think open_2 is a replacement for open() which does not imply that flag.

Reply:
@adammoody yeah good catch. I updated this.